phost/src/main.rs

```diff
@@ -404,29 +431,45 @@ async fn batch_sync_block(
         let r = req_sync_header(pr, &header_batch, authrotiy_change.as_ref()).await?;
         println!("  ..sync_header: {:?}", r);
 
-        let dispatch_window = batch_window - 1;
-        while !block_batch.is_empty() {
-            // TODO: fix the potential overflow here
-            let end_batch = block_batch.len() as isize - 1;
-            let batch_end = cmp::min(dispatch_window as isize, end_batch);
-            if batch_end >= 0 {
-                let mut dispatch_batch: Vec<BlockHeaderWithEvents> = block_batch
-                    .drain(..=(batch_end as usize))
-                    .map(|bwe| BlockHeaderWithEvents {
-                        block_header: bwe.block.block.header,
-                        events: bwe.events,
-                        proof: bwe.proof,
-                        worker_snapshot: None,
-                    })
-                    .collect();
-                maybe_take_worker_snapshot(client, events_decoder, &mut dispatch_batch).await?;
-                let r = req_dispatch_block(pr, &dispatch_batch).await?;
-                println!("  ..dispatch_block: {:?}", r);
-
-                // Update sync state
-                synced_blocks += dispatch_batch.len();
-            }
-        }
+        let para_fin_header = chain_client::get_parachain_heads(&client, &paraclient, Some(last_header_hash)).await;
```

Comment:
// TODO: pass header to TEE with storage proof to validate

Reply:
？